OneSignalSDK/onesignal/core/src/main/java/com/onesignal/user/IUserStateObserver.kt

```diff
@@ -0,0 +1,15 @@
+package com.onesignal.user
+
+/**
+ * A user state changed handler. Implement this interface and provide the implementation
```

Comment:
> A user state changed **observer**. Implement this interface and provide the implementation to be notified when the user **state** has changed.

Under the hood, the user has already changed in the SDK so this is a bit more correct to say it is the state object that changed.